helix-view/src/theme.rs

```diff
@@ -70,6 +97,58 @@ impl Loader {
             .unwrap_or_default()
     }
 
+    // merge one theme into the parent theme
+    fn merge_themes(&self, parent_theme_toml: Value, theme_toml: Value) -> Value {
+        let parent_palette = parent_theme_toml.get("palette");
+        let palette = theme_toml.get("palette");
+
+        // handle the table seperately since it needs a `merge_depth` of 2
+        // this would conflict with the rest of the theme merge strategy
+        let palette_values = if let Some(parent_palette) = parent_palette {
+            if let Some(palette) = palette {
+                merge_toml_values(parent_palette.clone(), palette.clone(), 2)
+            } else {
+                parent_palette.clone()
+            }
+        } else {
+            if let Some(palette) = palette {
+                palette.clone()
+            } else {
+                Map::new().into()
+            }
+        };
```

Comment:
This is a little cleaner as a `match`

```suggestion
        let palette_values = match (parent_palette, palette) {
            (Some(parent_palette), Some(palette)) => {
                merge_toml_values(parent_palette.clone(), palette.clone(), 2)
            }
            (Some(parent_palette), None) => parent_palette.clone(),
            (None, Some(palette)) => palette.clone(),
            (None, None) => Map::new().into(),
        };
```

Reply:
You are right of course, was only thinking about let chains and this easy and clean solution didn't come to mind.